scripts/init.sh

```diff
@@ -25,7 +25,8 @@ done
 echo "Configure rp_filter loose... "
 TOKEN=$(curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 60")
 HOST_IP=$(curl -H "X-aws-ec2-metadata-token: $TOKEN" http://169.254.169.254/latest/meta-data/local-ipv4)
-PRIMARY_IF=$(ip -4 -o a | grep "$HOST_IP/" | awk '{print $2}')
+PRIMARY_MAC=$(curl -H "X-aws-ec2-metadata-token: $TOKEN" http://169.254.169.254/latest/meta-data/mac)
```

Comment:
Is it worth taking the dependency on instance metadata? Didn't we in the past experience some race conditions around instance metadata access? So, in this case what happens if the initial curl to get the PRIMARY_MAC fails? If we want to go down this route, should we add retries for failure scenarios?

Reply:
I assume the past issues with metadata are not for primary ENI and it is already used for token and host_ip. @nithu0115 any comments on this?

Reply:
I agree with @achevuru here. I have seen issues where IMDS can take up to 30s to provide the information of a new ENI after fetching the token. We should add retries to cover such unexpected failures.

Reply:
But say if metadata isn't updated then PRIMARY_IF will be null and following command will fail. Kubelet will try restart aws-node. Adding retry will be faster but do you think kubelet retry doesn't suffice this use case?

```
PRIMARY_IF=$(ip -o link show | grep -F "link/ether $PRIMARY_MAC" | awk -F'[ :]+' '{print $2}')
 sysctl -w "net.ipv4.conf.$PRIMARY_IF.rp_filter=2"
```

Reply:
@achevuru and @nithu0115 - I have added the retry for failure cases.

Reply:
Sure, but I believe it might be better if we handle that with in the init script. I mean, if one sees a bunch of restarts  that might invite questions/issues on what is causing the restarts and if it indicates a problem.

Reply:
Yeah makes sense.